traffic_prophet/countmatch/reader.py

```diff
@@ -300,24 +229,26 @@ def get_zipreader(self, zipname):
                            'data': data,
                            'year': data.at[0, 'Timestamp'].year}
 
-    def read_sql(self):
-        """Read PostgreSQL table contents into AnnualCount objects."""
-        # ptcs and sttcs hold arrays of processed counts.
-        ptcs = {}
-        sttcs = {}
 
+class ReaderPostgres(ReaderBase):
+
+    def read_source(self, counts):
+        """Read PostgreSQL table contents into RawAnnualCount objects."""
         # Cycle through all relevant years.
         for year in range(cfg.cm['min_year'], cfg.cm['max_year'] + 1):
-            current_counts = [AnnualCount.from_raw_data(c)
-                              for c in self.get_sqlreader(year)]
+            current_counts = [
+                RawAnnualCount.from_raw_data(
+                    self.preprocess_count_data(c))
+                for c in self.get_sqlreader(year)]
```

Comment:
It's fine that this is not idempotent, since the SQL reader can't be retrieved by the calling function.